# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/ScreenHelperGTK.cpp
# Commit: 94573ef77f8e
# Full Hash: 94573ef77f8e05a83cf463ed6f78798683542d25
# Author: stransky <stransky@redhat.com>
# Date: 2025-07-11 21:24:55
# Description:
#   Bug 1976669 [Wayland] Check mScreenGetter before use r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D256874
# ==============================================================================

diff -r 5b147ee80018 -r 94573ef77f8e widget/gtk/ScreenHelperGTK.cpp
--- a/widget/gtk/ScreenHelperGTK.cpp	Fri Jul 11 11:48:05 2025 +0000
+++ b/widget/gtk/ScreenHelperGTK.cpp	Fri Jul 11 12:01:00 2025 +0000
@@ -406,7 +406,9 @@
 
 void WaylandMonitor::ImageDescriptionDone() {
   LOG_SCREEN("WaylandMonitor()[%p] ImageDescriptionDone", this);
-  mScreenGetter->AddScreen(MakeScreenGtk(mMonitor, mIsHDR));
+  if (mScreenGetter) {
+    mScreenGetter->AddScreen(MakeScreenGtk(mMonitor, mIsHDR));
+  }
 }
 
 void WaylandMonitor::ImageDescriptionReady() {
